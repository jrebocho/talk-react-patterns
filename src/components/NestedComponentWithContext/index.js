import React, { Component } from 'react'
import './NestedComponentWithContext.css'
import Stateless from '../Stateless'

const ToggleSwitchContext = React.createContext()

class NestedComponentWithContext extends Component {
  constructor(props) {
    super(props)
    this.state = { on: false }
    this.toggle = this.toggle.bind(this)
  }

  toggle() {
    this.setState(prevState => ({
      on: !prevState.on
    }))
  }

  render() {
    const boxColor = this.state.on ? 'green' : 'red'

    return (
      <ToggleSwitchContext.Provider value={this.state}>
        <div className={`outer-box ${boxColor}`} onClick={this.toggle}>
          <YellowBox />
        </div>
      </ToggleSwitchContext.Provider>
    )
  }
}

const YellowBox = () => (
  <div className="yellow">
    <ToggleSwitchContext.Consumer>
      {context => <Stateless on={context.on} />}
    </ToggleSwitchContext.Consumer>
  </div>
)

export default NestedComponentWithContext
