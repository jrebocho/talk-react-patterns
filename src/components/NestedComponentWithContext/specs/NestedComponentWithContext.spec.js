import React from 'react'
import { shallow } from 'enzyme'
import NestedComponentWithContext from '../../NestedComponentWithContext'

describe('<NestedComponentWithContext />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NestedComponentWithContext />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is clicked', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<NestedComponentWithContext />)
      wrapper.simulate('click')

      expect(wrapper).toMatchSnapshot()
    })
  })
})
