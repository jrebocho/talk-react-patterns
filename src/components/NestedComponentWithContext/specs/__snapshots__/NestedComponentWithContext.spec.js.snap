// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NestedComponentWithContext /> renders correctly 1`] = `
<[object Object]
  value={
    Object {
      "on": false,
    }
  }
>
  <div
    className="outer-box red"
    onClick={[Function]}
  >
    <YellowBox />
  </div>
</[object Object]>
`;

exports[`<NestedComponentWithContext /> when the component is clicked renders correctly 1`] = `
<[object Object]
  value={
    Object {
      "on": false,
    }
  }
>
  <div
    className="outer-box red"
    onClick={[Function]}
  >
    <YellowBox />
  </div>
</[object Object]>
`;
